src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -262,25 +267,25 @@ impl CloudwatchLogsSvc {
         })
     }
 
-    pub fn encode_log(&self, mut log: LogEvent) -> InputLogEvent {
-        let timestamp =
-            if let Some(Value::Timestamp(ts)) = log.remove(&event::log_schema().timestamp_key()) {
-                ts.timestamp_millis()
-            } else {
-                Utc::now().timestamp_millis()
-            };
+    pub fn encode_log(&self, mut log: LogEvent) -> Option<InputLogEvent> {
+        let timestamp = match log.remove(&event::log_schema().timestamp_key()) {
+            Some(Value::Timestamp(ts)) => ts.timestamp_millis(),
+            _ => Utc::now().timestamp_millis(),
+        };
 
-        match self.encoding.codec() {
-            Encoding::Json => {
-                let message = serde_json::to_string(&log).unwrap();
-                InputLogEvent { message, timestamp }
-            }
-            Encoding::Text => {
-                let message = log
-                    .get(&event::log_schema().message_key())
-                    .map(|v| v.to_string_lossy())
-                    .unwrap_or_else(|| "".into());
-                InputLogEvent { message, timestamp }
+        let message = match self.encoding.codec() {
+            Encoding::Json => serde_json::to_string(&log).unwrap(),
+            Encoding::Text => log
+                .get(&event::log_schema().message_key())
+                .map(|v| v.to_string_lossy())
+                .unwrap_or_else(|| "".into()),
+        };
+
+        match message.len() {
+            length if length <= MAX_MESSAGE_SIZE => Some(InputLogEvent { message, timestamp }),
+            length => {
+                error!(message = "Message exceeds CloudWatch maximum size", %length);
```

Comment:
Might consider rate limiting this?

Reply:
Another option might be to return a result from this function?

So other places would have like:

```diff
-            .map(|e| self.encode_log(e))
+            .filter_map(|e| self.encode_log(e).map_err(error!(...)).ok())
```